### PR TITLE
CQL: Proxy protocol integration tests

### PR DESCRIPTION
**Pull request by mpenick**
Fixes https://github.com/stargate/stargate/issues/322

### CODE REVIEW

File: persistence-test/src/main/java/io/stargate/it/storage/IsDseCondition.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * A condition that checks if the persistence backend uses DSE.
+ *
+ * <p>The test/suite is skipped if the persistence backend is not running DSE.
+ */
+public class IsDseCondition implements ExecutionCondition {
+  @Override
+  public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext extensionContext) {
+    if (CcmBridge.DSE_ENABLEMENT) {
```
Comment (dimas-b) [inline persistence-test/src/main/java/io/stargate/it/storage/IsDseCondition.java:16]: Could we get the current cluster (the same way `StargateContainer` gets it) and use `ClusterConnectionInfo.isDse()` instead? I think avoiding static references would be nice in this context.

File: testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java [hunk 2]
```diff
@@ -53,4 +53,8 @@ public boolean isCassandra4() {
     return !backend.isDse()
         && Version.parse(backend.clusterVersion()).nextStable().compareTo(Version.V4_0_0) >= 0;
   }
+
+  public boolean isDse() {
```
Comment (dimas-b) [inline testing/src/test/java/io/stargate/it/BaseOsgiIntegrationTest.java:57]: Is this method called?

File: testing/src/test/java/io/stargate/it/driver/CqlSessionSpec.java [hunk 3]
```diff
@@ -84,4 +84,6 @@
    * #initQueries()} are ignored.
    */
   boolean createSession() default true;
+
+  boolean noDefaultContactPoints() default false;
```
Comment (dimas-b) [inline testing/src/test/java/io/stargate/it/driver/CqlSessionSpec.java:88]: `boolean useDefaultContactPoints() default true;`?

File: testing/src/test/java/io/stargate/it/proxy/ProxyDnsCondition.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    ProxySpec proxySpec = maybeProxySpec.orElse(ProxyExtension.DEFAULT_PROXY_SPEC);
+
+    List<InetAddress> resolvedAddresses;
+    try {
+      resolvedAddresses =
+          Arrays.stream(InetAddress.getAllByName(proxySpec.verifyProxyDnsName()))
+              .map(a -> getByName(a.getHostAddress()))
+              .collect(Collectors.toList());
+    } catch (UnknownHostException e) {
+      throw new UncheckedIOException("Unable to determine addresses for proxy DNS", e);
```
Comment (dimas-b) [inline testing/src/test/java/io/stargate/it/proxy/ProxyDnsCondition.java:49]: Could this break the test as opposed to disabling it?

Comment (mpenick) [reply testing/src/test/java/io/stargate/it/proxy/ProxyDnsCondition.java:49]: Good call. Fixing.

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**